samza-core/src/main/java/org/apache/samza/operators/MessageStreamImpl.java

```diff
@@ -113,18 +114,22 @@ public MessageStreamImpl(StreamApplicationDescriptorImpl streamAppDesc, Operator
   }
 
   @Override
-  public void sink(SinkFunction<? super M> sinkFn) {
+=======
+>>>>>>> b22baf9a6b35e5cb0b190be52f8a93cbc9c086cf
+  public MessageStream<M> sink(SinkFunction<? super M> sinkFn) {
     String opId = this.streamAppDesc.getNextOpId(OpCode.SINK);
     SinkOperatorSpec<M> op = OperatorSpecs.createSinkOperatorSpec(sinkFn, opId);
     this.operatorSpec.registerNextOperatorSpec(op);
+    return new MessageStreamImpl<>(this.streamAppDesc, op);
```

Comment:
Why new message stream instead of returning 'this'?

Reply:
On second thought, returning new MessageStream will be required to get the correct chaining behavior, such that the next operator runs after this one, not 'in parallel'. Is that right?